ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -744,3 +755,64 @@ waitForSystemToQuiesce st@PeerSelectionState{
                                       }
         }
   | otherwise = GuardedSkip Nothing
+
+-- |This job, which is initiated by monitorLedgerStateJudgement job,
+-- verifies whether the provided big ledger pools match up with the
+-- ledger state once the node catches up to the slot at which the
+-- snapshot was ostensibly taken
+--
+jobVerifyPeerSnapshot :: ( MonadSTM m
+                         , MonadThrow m)
+                      => LedgerPeerSnapshot
+                      -> LedgerPeersConsensusInterface m
+                      -> Job () m (Completion m peeraddr peerconn)
+jobVerifyPeerSnapshot (LedgerPeerSnapshot (slot, relays))
+                      LedgerPeersConsensusInterface {
+                        lpGetLatestSlot,
+                        lpGetLedgerPeers }
+  = Job job
+        throwIO
+        ()
+        "jobVerifyPeerSnapshot"
+  where
+    completion queueJob trace = return . Completion $ \st _now ->
+      Decision {
+        decisionTrace = maybeToList trace,
+        decisionState = st,
+        decisionJobs  = maybeToList queueJob }
+
+    job = do
+      result <-
+        atomically $ do
+          check . (slot >=) =<< lpGetLatestSlot
```

Comment:
This should be in the other part, when `jobVerifyPeerSnapshot` is called.

Reply:
I didn't like that the governor would be woken up every time the slot changed and then do it's various checks in the main loop when most of the time there'd be nothing to do but wait. So I moved this check here - there is no race because lpGetLatestSlot is not read anywhere else in the governor code. I think this is a better way.